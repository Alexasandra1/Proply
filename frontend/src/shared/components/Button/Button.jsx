import React, { useState } from "react"
import './Button.scss'


// export function Button() {
export function Button(props) {
    function handleClick() {
        if (props.onClick) {
            props.onClick();
        }
    }

    return (
        <div className="button">
            <button
                className={`button__button ${props.styleButton}`}
                onClick={handleClick}
            >
                {props.wordButton}
            </button>
        </div>
    );
}