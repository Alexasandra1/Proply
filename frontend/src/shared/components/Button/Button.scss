.styleButtonAboutUs {
    max-width: 24.75rem;
    height: 4.375rem;
    border: none;
    border-radius: 1.25rem;
    background: rgba(217, 217, 217, 0.40);
    font-family: Montserrat;
    font-size: 1.5rem;
    font-style: normal;
    font-weight: 600;
    margin-top: 1.44rem;
}

.logInPage__body__form__button,
.registrationPage__body__form__button,
.informationHouse__button,
.addHousePage__body__form__button {
    background: #2D7C83;
    width: 16.125rem;
    height: 4.625rem;
    color: rgba(255, 255, 255, 0.99);
    text-align: center;
    font-family: Montserrat;
    font-size: 2rem;
    font-style: normal;
    font-weight: 700;
    border: none;
}


.forgotButton{
    background: #2D7C83;
    width: 100%;
    height: 4.625rem;
    color: rgba(255, 255, 255, 0.99);
    text-align: center;
    font-family: Montserrat;
    font-size: 2rem;
    font-style: normal;
    font-weight: 700;
    border: none;
}

.informationHouse__button__delete{
    background-color: transparent;
    color: rgba(138, 0, 0, 0.99);
    text-align: center;
    font-family: Montserrat;
    font-size: 32px;
    border: 2px solid rgb(167, 0, 0);
    width: auto;
    height: 74px;
}

.profilePage__main__container__buttonInformation,
.logOutButton{
    background: #2D7C83;
    width: auto;
    height: auto;
    color: rgba(255, 255, 255, 0.99);
    text-align: center;
    font-family: Montserrat;
    font-size: 2rem;
    font-weight: 700;
    border: none;
    margin-bottom: 1%;
}

.logOutButton{
color: rgb(36, 26, 26);
}