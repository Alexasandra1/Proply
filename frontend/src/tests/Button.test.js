import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../shared/components/Button/Button';

test('renders correctly', () => {
  const props = {
    styleButton: 'test-style',
    wordButton: 'Test Button',
    onClick: jest.fn(),
  };

  const tree = renderer
    .create(<Button {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
