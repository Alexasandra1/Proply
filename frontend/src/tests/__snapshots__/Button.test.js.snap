// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="button"
>
  <button
    className="button__button test-style"
    onClick={[Function]}
  >
    Test Button
  </button>
</div>
`;
